### DISCUSSION TITLE
could i use curl/wget to download file from minio?

### DISCUSSION THREAD

**Discussion by furynerd** [minio/minio#20105] [Q&A]
hi, is it possible to download file from minio by curl/wget (because my use case need this)?

i got a `credentials.json`  indicates me something like `accessKey`, `secretKey`, `api:s3v4`.

i tried something like
```shell
curl -o mc -X GET \
    -H "Date: `date -R`" \
    -H "Authorization: s3v4 ${accessKey}:${secretKey}" \
    "http://ip.of.local.lan:port/bucket/dir/file"
```

but apparently didnt work as well as other simliar attemps...

appreciate for any advice.

thanks

**Comment by NotSoWell** [comment]
Yes it is possible - download Postman (great tool for testing http requests).

"Build" your request, test it by clicking "Send" button, if its ok u can click on "Code" button - it will show u a curl (or other code snippets even for different languages).

But u need to chose Authorization - "AWS Signature".

About Authorization header - auth in AWS S3 works different - it's not just appended **accesKey** and **secretKey**. MinIO supports (according to documentation https://min.io/docs/minio/linux/administration/identity-access-management.html) [AWS Signature Version 4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) and deprecated [Signature Version 2](https://docs.aws.amazon.com/AmazonS3/latest/userguide/auth-request-sig-v2.html) - you need to read through it, to understand how to build Authorization header.

![image](https://github.com/user-attachments/assets/08a3d6a4-8383-4db9-a8f1-4516e37fd5c5)
